### PR TITLE
adding masahaner dataset

### PR DESCRIPTION
**Pull request by dadelani**
Adding Masakhane dataset https://github.com/masakhane-io/masakhane-ner 

@lhoestq , can you please review

### CODE REVIEW

File: datasets/masakhaner/README.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+source_datasets:
+- original
+task_categories:
+- structure-prediction
+task_ids:
+- named-entity-recognition
+paperswithcode_id: null
+---
+
+# Dataset Card Creation Guide
```
Comment (lhoestq) [inline datasets/masakhaner/README.md:43]:
```suggestion
# Dataset Card for MasakhaNER
```

File: datasets/masakhaner/masakhaner.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                    # Masakhaner tokens are space separated
+                    splits = line.split(" ")
+                    tokens.append(splits[0])
+                    ner_tags.append(splits[1].rstrip())
+            # last example
+            yield guid, {
+                "id": str(guid),
+                "tokens": tokens,
+                "ner_tags": ner_tags,
+            }
```
Comment (lhoestq) [inline datasets/masakhaner/masakhaner.py:172]: Maybe check that the last example is not empty ?
```suggestion
            if tokens:
                yield guid, {
                    "id": str(guid),
                    "tokens": tokens,
                    "ner_tags": ner_tags,
                }
```

**Comment by dadelani** [comment]
Thank you for the review.

**Review by lhoestq — COMMENTED** [review]
Thanks for adding this dataset !

I just added a few comments

Also the CI is failing for two reasons:
1. you need to run `make style` to format the code
2. the dummy data are missing. They're used to quickly test that the script works as expected. You can generate them automatically using this command:
```
datasets-cli dummy_data datasets/masaknaner --auto_generate --n_lines 20
```

For more infos regarding the dummy data generation, please read: https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#automatically-add-code-metadata

Ideally the resulting dummy_data.zip files should be less than 20KB each, let me know if this is not the case

**Comment by dadelani** [comment]
Thanks a lot for the corrections and comments. 

I have resolved point 2. The make style still throws some errors, please see below

black --line-length 119 --target-version py36 tests src benchmarks datasets/**/*.py metrics
/bin/sh: 1: black: not found
Makefile:13: recipe for target 'style' failed
make: *** [style] Error 127

Can you help to resolve this?

**Comment by dadelani** [comment]
Thank you very much @lhoestq for the help.

**Review by lhoestq — APPROVED** [review]
This looks all good now !

Thanks a lot :)

**Pull request merged**